### PR TITLE
CORTX-28033: Updated Feature Endpoint Mapping 

### PR DESCRIPTION
**Pull request by abhijit1patil**
# Problem Statement
- [Jira ticket](https://jts.seagate.com/browse/CORTX-28033)
- For add Remove access key and  add Remove Caps, Link/Unlink Bucket  operations

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Deployment

- [Custom Build for this branch](https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5417/)
- [Deployment Job with Custom Build](https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/1257/)
- Cluster Deployed on the VM
image

# Dev Testing

- Create User
image

- Add Access Key 
image

- Add user caps
image

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]
Please follow PR - naming conventions.

File: schema/feature_endpoint_mapping.json [hunk 1]
```diff
@@ -264,15 +264,33 @@
...
+  },
+  "/api/*/s3/iam/keys": {
+    "feature_name": "s3_iam_user",
     "dependent_on": [
-      "rgw_s3"
+      "s3"
+    ]
+  },
+  "/api/*/s3/iam/caps/*": {
+    "feature_name": "s3_iam_user",
```
Comment (pawankumarsrivastava) [inline schema/feature_endpoint_mapping.json:285]: s3_iam_user -> iam_user_mgmt

File: schema/feature_endpoint_mapping.json [hunk 2]
```diff
@@ -264,15 +264,33 @@
...
   },
   "/api/*/s3/iam/users/*": {
-    "feature_name": "s3_iam_users",
+    "feature_name": "s3_iam_user",
+    "dependent_on": [
+      "s3"
+    ]
+  },
+  "/api/*/s3/iam/keys": {
+    "feature_name": "s3_iam_user",
```
Comment (pawankumarsrivastava) [inline schema/feature_endpoint_mapping.json:279]: s3_iam_user -> iam_user_mgmt

File: schema/feature_endpoint_mapping.json [hunk 3]
```diff
@@ -264,15 +264,33 @@
...
-    "feature_name": "s3_iam_users",
+    "feature_name": "s3_iam_user",
     "dependent_on": [
-      "rgw_s3"
+      "s3"
     ]
   },
   "/api/*/s3/iam/users/*": {
-    "feature_name": "s3_iam_users",
+    "feature_name": "s3_iam_user",
```
Comment (pawankumarsrivastava) [inline schema/feature_endpoint_mapping.json:273]: s3_iam_user -> iam_user_mgmt

File: schema/feature_endpoint_mapping.json [hunk 4]
```diff
@@ -264,15 +264,33 @@
     "dependent_on": []
   },
   "/api/*/s3/iam/users": {
-    "feature_name": "s3_iam_users",
+    "feature_name": "s3_iam_user",
```
Comment (pawankumarsrivastava) [inline schema/feature_endpoint_mapping.json:267]: s3_iam_user -> iam_user_mgmt

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: schema/feature_endpoint_mapping.json [hunk 5]
```diff
@@ -264,15 +264,33 @@
...
+    ]
+  },
+  "/api/*/s3/iam/caps/*": {
+    "feature_name": "s3_iam_user",
+    "dependent_on": [
+      "s3"
+    ]
+  },
+  "/api/*/s3/iam/bucket": {
+    "feature_name": "s3_bucket",
```
[the remark this answers is not in the document]
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:291]: updated

File: schema/feature_endpoint_mapping.json (hunk 1 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:285]: updated

File: schema/feature_endpoint_mapping.json (hunk 2 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:279]: updated

File: schema/feature_endpoint_mapping.json (hunk 3 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:273]: updated

File: schema/feature_endpoint_mapping.json (hunk 4 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:267]: updated

**Comment by abhijit1patil** [comment]
> Please follow PR - naming conventions.

Updated

File: schema/feature_endpoint_mapping.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (pawankumarsrivastava) [reply schema/feature_endpoint_mapping.json]: Please revert the changes to the original one.

File: schema/feature_endpoint_mapping.json (hunk 1 above)
Comment (pawankumarsrivastava) [reply schema/feature_endpoint_mapping.json]: Please revert the changes to the original one.

File: schema/feature_endpoint_mapping.json (hunk 2 above)
Comment (pawankumarsrivastava) [reply schema/feature_endpoint_mapping.json]: Please revert the changes to the original one.

File: schema/feature_endpoint_mapping.json (hunk 3 above)
Comment (pawankumarsrivastava) [reply schema/feature_endpoint_mapping.json]: Please revert the changes to the original one.

File: schema/feature_endpoint_mapping.json (hunk 4 above)
Comment (pawankumarsrivastava) [reply schema/feature_endpoint_mapping.json]: Please revert the changes to the original one.

File: schema/feature_endpoint_mapping.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:291]: reverted

File: schema/feature_endpoint_mapping.json (hunk 1 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:285]: reverted

File: schema/feature_endpoint_mapping.json (hunk 2 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:279]: revereted

File: schema/feature_endpoint_mapping.json (hunk 3 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:273]: reverted

File: schema/feature_endpoint_mapping.json (hunk 4 above)
Comment (abhijit1patil) [reply schema/feature_endpoint_mapping.json:267]: reverted

**Review by rohitkolapkar — APPROVED** [review]

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**